pymeasure/display/manager.py

```diff
@@ -260,7 +260,13 @@ def _finish(self):
         experiment.browser_item.setProgress(100.)
         for curve in experiment.curve_list:
             if curve:
-                curve.update_data()
+                # For single plots, update_data() on ResultsCurve
+                if type(curve) == ResultsCurve:
+                    curve.update_data()
+                # For multiple plots, update_data() on list of ResultsCurve
+                elif type(curve) == list:
+                    for c in curve:
+                        c.update_data()
```

Comment:
To me, it would seem nicer if the curve_list would consist of only actual ResultsCurve objects, not a mix of ResultsCurves and Lists of ResultsCurves, either by expanding the list before it is committed (in the `new_experiment` method of the ManagedWindow(Base) class), or alternatively by introducing a new type of Curve, the MultiResultsCurve.